pkg/pillar/canbus/canbus.go

```diff
@@ -0,0 +1,881 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// CAN Bus support package
+// This package provides functions to setup physical and/or create virtual
+// CAN interfaces
+package canbus
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netlink/nl"
+	"golang.org/x/sys/unix"
+)
+
+// Types of CAN interfaces
+const (
+	// LinkVCAN Virtual CAN interface type
+	LinkVCAN = "vcan"
+	// LinkCAN Physical CAN interface type
+	LinkCAN = "can"
+)
+
+// The following constants were taken from Linux's kernel header:
+// include/uapi/linux/can/netlink.h (version 6.6)
+// PS: All of them are treat as uint32 in netlink packages
+const (
+	IFLA_CAN_TDC          = unix.IFLA_CAN_BITRATE_MAX + 1
+	IFLA_CAN_TDC_TDCV_MIN = 0x1
+	IFLA_CAN_TDC_TDCV_MAX = 0x2
+	IFLA_CAN_TDC_TDCO_MIN = 0x3
+	IFLA_CAN_TDC_TDCO_MAX = 0x4
+	IFLA_CAN_TDC_TDCF_MIN = 0x5
+	IFLA_CAN_TDC_TDCF_MAX = 0x6
```

Comment:
would be nice to have a comment here what those abbreviations mean, like TDC or TDCF

Reply:
Add the relevant comment for TDC meaning.